lib/battle_snake/snake.ex

```diff
@@ -1,5 +1,12 @@
+defimpl Poison.Encoder, for: BattleSnake.Snake do
+  def encode(snake, opts) do
+    keys = [:coords, :id, :taunt, :health_points, :name] ++
+      if Keyword.get(opts, :consumer) do  [ :head_url, :color] else [] end
+    Poison.encode!(Map.take(snake, keys))
+  end
+end
+
```

Comment:
I would strongly prefer that any additional rendering "modes" are kept as completely isolated behaviour. My issue before was that the behaviour changed the standard control flow to use what was previously defined as a secondary mode when an option was absent.

My issue with this is that it is now tightly coupled with the normal control flow of this method, should I make modifications to this this method for the rest of the application it will break compatibility with any consumer of this endpoint.

What I would like from this method is that I can change anything about it as I please, and it not have it affect anything that relies on this addition.  

Something along the lines of this, if you really need to preserve options:

```elixir
def encode(snake, opts) do
  case Keyword.pop(opts, :mode) do
    {:consumer, opts} ->
      consumer_encode(snake, opts)

    {_, opts} ->
      do_encode(snake, opts)
  end
end

def do_encode(snake, opts) do
  keys = [:coords, :id, :taunt, :health_points, :name]
  snake
  |> Map.take(keys)
  |> Poison.encode!(opts)
end

def consumer_encode(snake, opts) do
  keys = [:coords, :id, :taunt, :health_points, :name, :head_url, :color]
  snake
  |> Map.take(keys)
  |> Poison.encode!(opts)
end
```